substrate-node/pallets/pallet-smart-contract/src/types.rs

```diff
@@ -47,24 +63,36 @@ impl<T: Config> Contract<T> {
 
     pub fn get_node_id(&self) -> u32 {
         match self.contract_type.clone() {
-            ContractData::RentContract(c) => c.node_id,
-            ContractData::NodeContract(c) => c.node_id,
+            ContractData::CapacityReservationContract(c) => c.node_id,
+            ContractData::DeploymentContract(_) => 0,
             ContractData::NameContract(_) => 0,
         }
     }
 }
 
-#[derive(Clone, Eq, PartialEq, RuntimeDebugNoBound, Encode, Decode, TypeInfo, MaxEncodedLen)]
+#[derive(Clone, Eq, PartialEq, RuntimeDebugNoBound, Encode, Decode, TypeInfo)]
 #[scale_info(skip_type_params(T))]
 #[codec(mel_bound())]
-pub struct NodeContract<T: Config> {
+pub struct CapacityReservationContract {
     pub node_id: u32,
+    pub resources: ConsumableResources,
+    pub group_id: Option<u32>,
+    pub public_ips: u32,
+    pub deployment_contracts: Vec<u64>,
+}
+
+#[derive(Clone, Eq, PartialEq, RuntimeDebugNoBound, Encode, Decode, TypeInfo, MaxEncodedLen)]
+#[scale_info(skip_type_params(T))]
+#[codec(mel_bound())]
+pub struct DeploymentContract<T: Config> {
+    pub capacity_reservation_id: u64,
     // Hash of the deployment, set by the user
     // Max 32 bytes
     pub deployment_hash: DeploymentHash,
     pub deployment_data: BoundedVec<u8, MaxDeploymentDataLength<T>>,
     pub public_ips: u32,
```

Comment:
Can't we remove this `public_ips` field? Since it's also saved on the capacity reservation contract, keeping the list here is fine.

Reply:
Sure. Why do we need the public_ips_list? Does zos need it?

Reply:
Deprecating this comment